libpod/container_internal.go

```diff
@@ -781,12 +780,6 @@ func (c *Container) mountStorage() (err error) {
 
 // cleanupStorage unmounts and cleans up the container's root filesystem
 func (c *Container) cleanupStorage() error {
-	if !c.state.Mounted {
-		// Already unmounted, do nothing
-		logrus.Debugf("Storage is already unmounted, skipping...")
-		return nil
-	}
-
 	for _, mount := range c.config.Mounts {
 		if err := c.unmountSHM(mount); err != nil {
```

Comment:
Is this safe if we are not c.state.Mounted?

Reply:
We would end up with warnings. but It should be only done if mounted.